src/Integration.UnitTests/SLCore/SLCoreConstantsProviderTests.cs

```diff
@@ -28,145 +31,188 @@
+        var testSubject = CreateTestSubject(ideName: ideName);
+        var expectedClientConstants = new ClientConstantsDto(ideName,
+            $"SonarLint Visual Studio/{VersionHelper.SonarLintVersion}",
+            Process.GetCurrentProcess().Id);
+        var actual = testSubject.ClientConstants;
+
+        actual.Should().BeEquivalentTo(expectedClientConstants);
+    }
 
-        [TestMethod]
-        public void ClientConstants_ShouldBeExpected()
-        {
-            const string ideName = "MyIde";
-            SetupIdeName(ideName);
+    [TestMethod]
+    public void ClientConstants_VsOperationCalledOnlyOnce()
+    {
+        var testSubject = CreateTestSubject(out var vsShell, ideName: "somename");
 
-            var expectedClientConstants = new ClientConstantsDto(ideName, $"SonarLint Visual Studio/{VersionHelper.SonarLintVersion}",
-                Process.GetCurrentProcess().Id);
-            var actual = testSubject.ClientConstants;
+        _ = testSubject.ClientConstants;
+        _ = testSubject.ClientConstants;
+        _ = testSubject.ClientConstants;
+        _ = testSubject.ClientConstants;
 
-            actual.Should().BeEquivalentTo(expectedClientConstants);
-        }
+        vsShell.ReceivedWithAnyArgs(1).GetProperty(default, out _);
+    }
 
-        [TestMethod]
-        public void ClientConstants_VsOperationCalledOnlyOnce()
-        {
-            SetupIdeName("somename");
+    [TestMethod]
+    public void FeatureFlags_ShouldBeExpected()
+    {
+        var testSubject = CreateTestSubject();
+        var expectedFeatureFlags = new FeatureFlagsDto(true, true, true, true, false, false, true, false);
+        var actual = testSubject.FeatureFlags;
 
-            _ = testSubject.ClientConstants;
-            _ = testSubject.ClientConstants;
-            _ = testSubject.ClientConstants;
-            _ = testSubject.ClientConstants;
+        actual.Should().BeEquivalentTo(expectedFeatureFlags);
+    }
 
-            vsShell.ReceivedWithAnyArgs(1).GetProperty(default, out _);
-        }
+    [TestMethod]
+    public void TelemetryConstants_ShouldBeExpected()
+    {
+        var versionProvider = Substitute.For<IVsVersionProvider>();
+        var version = Substitute.For<IVsVersion>();
+        version.DisplayName.Returns("Visual Studio Professional 2022");
+        version.InstallationVersion.Returns("17.10.55645.41");
+        version.DisplayVersion.Returns("17.10.0 Preview 3.0");
+        versionProvider.Version.Returns(version);
+
+        var testSubject = CreateTestSubject(versionProvider: versionProvider);
+        VisualStudioHelpers.VisualStudioVersion = "1.2.3.4";
+        var expectedString = $$"""
+                         {
+                           "productKey": "visualstudio",
+                           "productName": "SonarLint Visual Studio",
+                           "productVersion": "{{typeof(TelemetryTimer).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>().Version}}",
```

Comment:
`TelemetryTimer` is going to be deleted, use a different class

Reply:
https://github.com/SonarSource/sonarlint-visualstudio/pull/5531/files#diff-1981c848372dfb6ebd8e3346e790c002216a833fac2332807552c9feeab9237dR36